### ISSUE TITLE
New APIs: Address Validation, Datasets, Places, Routing

### ISSUE THREAD

**Issue by wangela** [googlemaps/google-maps-services-js#1105]
If you are looking for Node.js client libraries for the following APIs, see the [Google Maps Platform APIs in the Cloud Client Libraries for Node.js](https://github.com/googleapis/google-cloud-node/tree/main/packages) ([releases](https://github.com/googleapis/google-cloud-node/releases?q=maps&expanded=true)).

- [Address Validation API](https://github.com/googleapis/google-cloud-node/tree/main/packages/google-maps-addressvalidation)
- [Datasets API](https://github.com/googleapis/google-cloud-node/tree/main/packages/google-maps-mapsplatformdatasets)
- [Places API (New)](https://github.com/googleapis/google-cloud-node/tree/main/packages/google-maps-places)
- [Routes API](https://github.com/googleapis/google-cloud-node/tree/main/packages/google-maps-routing)

The new APIs will not be added to this client library.

**Comment by devuxer** [comment]
@wangela, Thank you for this information, but I'm having trouble understanding what to do with the links you listed.

I am interested in using "Places API (New)" with via @googlemaps/google-maps-services-js. Are you saying we need a different npm package? If so, what is it called? If not, what should we do? Thank you!

**Comment by teresarwqin** [comment]
@devuxer You are right. You will need an npm package for Node.js client libraries.  Please see the section of [Installing the client library](https://github.com/googleapis/google-cloud-node/tree/main/packages/google-maps-places#installing-the-client-library)

And here is the [npm page](https://www.npmjs.com/package/@googlemaps/places)

**Comment by vladshcherbin** [comment]
Why is this information not pinned in readme top section ?

**Comment by LarsFlieger** [comment]
I'm also wondering why this information about these libraries isn't included somewhere in the [docs](https://developers.google.com/maps/documentation/javascript/libraries). Or am I missing something?

It's sometimes confusing trying to find the right libraries.

**Comment by daralthus** [comment]
then put a huge "DEPRECATED" in the readme!

**Comment by plebcity** [comment]
The old API's are going to be deprecated on the 1st of March 2025. New users won't be able to enable them anymore and existing user will have to pay when they previously didn't have to.
See: https://developers.google.com/maps/billing-and-pricing/faq?utm_medium=MSA&utm_source=GMP_Billing_Admin&utm_campaign=Dec_24&utm_content=FAQ

**Comment by nicolae536** [comment]
If I see correctly they will not update this package: And we have to migrate to the new packages... https://github.com/googleapis/google-cloud-node/tree/main/packages Great we used the package to not maintain api calls, now we basically switch packages instead of simple version bump...

**Comment by benbuzbee** [comment]
This is aweful. I wasted 2 hours not understanding why I wasnt authorized to use the old Google Places API, just to find its deprecated, and the new packages use some complex auth mechanism, that fails for me anyway with `Client network socket disconnected before secure TLS connection was established`  which feels like not my fault.

**Comment by teresarwqin** [comment]
> I'm also wondering why this information about these libraries isn't included somewhere in the [docs](https://developers.google.com/maps/documentation/javascript/libraries). Or am I missing something?
> 
> It's sometimes confusing trying to find the right libraries.

The doc you linked is the service libraries for you to import into JavaScript. For example, you will need to import the Places library when you need to use the places service in JavaScript.

The original post is about the client libraries to accelerate API request integration on server side. For example the Routes API client library - https://developers.google.com/maps/documentation/routes/client-libraries
